File: 프로그래머스/1/42748. K번째수/K번째수.py
# 1. 문자열 슬라이스
# 2. for문에서 커맨드 [0], [1] 까지 따로 뽑아서도 될듯


# arrar_str 생성
# 커맨드만큼 for문
#  arrar_str[i[0]-1:i[1]]
# 이거 다시 list 후 list[i[2]-1]처리

def solution(array, commands):
    answer = []
    
    for i in commands:
        arr = []
        arr = array[i[0]-1: i[1]]
     
        arr.sort()
        print(arr)
        answer.append(arr[i[2]-1])

    return answer

# 아 10이상부터는 슬라이스가 안되네 -> 배열도 슬라이스가 되는구나
# 오 여기서 맵

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    for(vector<int> v : commands){
        vector<int> imsi; // 실전에서는 어떤 변수명으로 할지
        imsi.assign(array.begin()+(v[0]-1), array.begin()+v[1]);
        sort(imsi.begin(), imsi.end());
        answer.push_back(imsi[v[2]-1]);
    }
    return answer;
}



// 1. commands size대로 for문
// 2. 새로운 배열 생성 후 커멘드 첫번째 두번째로 자른값 넣기 숫자-1 주의
// 3. 정렬
// 4. 숫자-1  answer에 넣기